modules/apps/static/portal-k8s-agent/portal-k8s-agent-impl/src/main/java/com/liferay/portal/k8s/agent/internal/LiferayRoutesEmitter.java

```diff
@@ -0,0 +1,509 @@
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.util.FileUtil;
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.PortalInetSocketAddressEventListener;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.util.PropsValues;
+
+import java.io.File;
+import java.io.IOException;
+
+import java.net.InetSocketAddress;
+
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Gregory Amerson
+ */
+@Component(
+	property = "service.ranking:Integer=-1",
+	service = PortalK8sConfigMapModifier.class
+)
+public class LiferayRoutesEmitter
+	implements PortalInetSocketAddressEventListener,
+			   PortalK8sConfigMapModifier {
+
+	@Override
+	public Result modifyConfigMap(
+		Consumer<PortalK8sConfigMapModifier.ConfigMapModel>
+			configMapModelConsumer,
+		String configMapName) {
+
+		Objects.requireNonNull(
+			configMapModelConsumer, "Config map model consumer is null");
+
+		Map<String, String> annotations = new TreeMap<>();
+		Map<String, String> binaryData = new TreeMap<>();
+		Map<String, String> data = new TreeMap<>();
+		Map<String, String> labels = new TreeMap<>();
+
+		configMapModelConsumer.accept(
+			new ConfigMapModel() {
+
+				@Override
+				public Map<String, String> annotations() {
+					return annotations;
+				}
+
+				@Override
+				public Map<String, String> binaryData() {
+					return binaryData;
+				}
+
+				@Override
+				public Map<String, String> data() {
+					return data;
+				}
+
+				@Override
+				public Map<String, String> labels() {
+					return labels;
+				}
+
+			});
+
+		if (binaryData.isEmpty() && data.isEmpty()) {
+			if (_log.isInfoEnabled()) {
+				_log.info(
+					StringBundler.concat(
+						"Config map does not exist and no data was supplied ",
+						"for ", configMapName, " resulting in no change"));
```

Comment:
I think this logging statement is off given we're deleting the config tree, no?

Reply:
Yeah that isn't right, fixing.